test/external/dataplane/connectivity.go

```diff
@@ -224,6 +225,8 @@ func testExternalConnectivity(p testParams) {
 		targetIP = podIP
 	case tcp.ServiceIP:
 		targetIP = svcIP
+		// Wait for the service to be ready.
+		time.Sleep(2 * time.Second)
```

Comment:
Adding a blind 2 sec sleep seems kludgy. Can we query the service to determine readiness?

Reply:
The other approach to fix it will be to retry the access for some time until it succeeds [here](https://github.com/submariner-io/submariner/pull/1763/files/bd38646d0a7f851d1da5db48710e25cbf786b8f7#diff-61634c0682ec746657beb82d32015cc6c7ca90694f9740bbbbbe4ac1bf12d579R236), but it also will be a big change, because current implementation of [`docker.RunCommand`](https://github.com/submariner-io/shipyard/blob/devel/test/e2e/framework/docker.go#L79) doesn't allow error. Can it be merged as a workaround and improve later?

Reply:
OK - that's fine. We need a `docker.RunCommandUntil` function.

Reply:
Please add a TODO comment so that we dont miss it.

Reply:
Added a TODO comment:

```golang
// TODO: Improve this by retrying access not a blind 2 sec sleep.
```

Reply:
> We need a docker.RunCommandUntil function.

Also started working on it, https://github.com/submariner-io/shipyard/issues/767#issue-1186666284 .